modules/subtitle-parser/src/main/java/org/opencastproject/subtitleparser/webvttparser/WebVTTParser.java

```diff
@@ -77,6 +77,11 @@ public WebVTTSubtitle parse(InputStream is) throws IOException, SubtitleParsingE
       throw new SubtitleParsingException("WEBVTT Header line is null");
     }
 
+    // Remove any optional UTF-8 Byte-Order-Marker (BOM)
+    if (line.startsWith("\uFEFF")) {
+      line = line.substring(1);
+    }
```

Comment:
Instead of a custom implementation that only handles the UTF-8 case, why not use a library? The `BOMInputStream` class from Apache Commons IO seems to do the trick.

Reply:
@mtneug would you update Apache Commons IO from 2.8.0 to 2.12.0 in Opencast or would you use the deprecated constructor

> [BOMInputStream](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/input/BOMInputStream.html#%3Cinit%3E(java.io.InputStream,boolean))([InputStream](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/io/InputStream.html) delegate, boolean include)